### PR TITLE
Added secret manager support for sqlserver-to-bigquery template

### PR DESCRIPTION
**Pull request by fayezinislam**

### CODE REVIEW

**Review by damccorm — COMMENTED** [review]
Thanks! Overall this looks good to me, had some minor comments

File: v2/sqlserver-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/options/SQLServerToBigQueryOptions.java [hunk 1]
```diff
@@ -24,15 +24,30 @@ public interface SQLServerToBigQueryOptions extends JdbcToBigQueryOptions {
...
       example = "jdbc:sqlserver://localhost;databaseName=sampledb")
   String getConnectionURL();
 
+  @TemplateParameter.Text(
+      optional = true,
+      regexes = {"(^.+$|projects/.*/secrets/.*/versions/.*)"},
+      description = "JDBC connection username.",
+      helpText =
+          "The username to use for the JDBC connection. You can pass in this value as a string that's encrypted with a Cloud KMS key and then Base64-encoded. Remove whitespace characters from the Base64-encoded string. Can be a Secret Manager secret in the form projects/{project}/secrets/{secret}/versions/{secret_version}.")
+  String getUsername();
```
[the remark this answers is not in the document]
Comment (fayezinislam) [reply v2/sqlserver-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/options/SQLServerToBigQueryOptions.java]: Done

File: v2/jdbc-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/JdbcToBigQuery.java [hunk 2]
```diff
@@ -202,4 +208,25 @@ static Write<TableRow> writeToBQTransform(JdbcToBigQueryOptions options) {
...
+
+  /**
+   * Retrieves a secret value from SecretManagerUtils if the input string matches the specified
+   * pattern.
+   *
+   * @param secret The input string representing a potential secret.
+   * @return The secret value if the input matches the pattern and the secret is found, otherwise
+   *     the original input string.
+   */
+  private static String maybeParseSecret(String secret) {
```
[the remark this answers is not in the document]
Comment (fayezinislam) [reply v2/jdbc-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/JdbcToBigQuery.java:220]: Done

**Review by damccorm — APPROVED** [review]
Thanks, this LGTM

**Comment by damccorm** [comment]
Running checks, will merge if they pass

**Comment by damccorm** [comment]
Looks like spotless is failing, could you please run `mvn spotless:apply`

**Comment by fayezinislam** [comment]
> Looks like spotless is failing, could you please run `mvn spotless:apply`

Fixed. Thank you!

File: v2/jdbc-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/JdbcToBigQueryIT.java [hunk 3]
```diff
@@ -419,4 +421,27 @@ private List<Map<String, Object>> getJdbcData(List<String> columns) {
...
+  @Test
+  void testMaybeParseSecret_invalidSecretPath_returnsOriginalString() throws Exception {
+    String secretPath = "not-a-secret";
+    String expectedSecret = "not-a-secret";
+    String actualSecret = invokeMaybeParseSecret(secretPath);
+    assertEquals(expectedSecret, actualSecret);
+  }
+
+  @Test
+  void testMaybeParseSecret_nullInput_returnsNull() throws Exception {
```
Comment (damccorm) [inline v2/jdbc-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/JdbcToBigQueryIT.java:442]: Looks like these are causing failures:

```
Invalid test class 'com.google.cloud.teleport.v2.templates.JdbcToBigQueryIT':
  1. Method testMaybeParseSecret_invalidSecretPath_returnsOriginalString() should be public
  2. Method testMaybeParseSecret_nullInput_returnsNull() should be public
```

From https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/runs/11448082743?pr=1947

Comment (fayezinislam) [reply v2/jdbc-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/JdbcToBigQueryIT.java]: Done. Changed to public.

**Review by damccorm — APPROVED** [review]
Thanks!

**Comment by fayezinislam** [comment]
Thanks for approving the merge request. When can we expect this to be available in Dataflow Production through the GCP console?

**Comment by damccorm** [comment]
The exact ETA can vary, but typically, we target ~2 weeks from merged to fully rolled out (if there are no issues found while rolling out).

**Comment by fayezinislam** [comment]
Fantastic. Thanks again for your support through out this process.

**Pull request merged**